pkg/checker/version/version.go

```diff
@@ -147,7 +146,9 @@ func latestSemver(opts *api.Options, tags []api.ImageTag) (*api.ImageTag, error)
 			// If the latest set is less than
 			latestV.LessThan(v) ||
 			// If the latest is the same tag, but smaller timestamp
-			(latestV.Equal(v) && tags[i].Timestamp.After(latestImageTag.Timestamp)) {
+			(latestV.Equal(v) && tags[i].Timestamp.After(latestImageTag.Timestamp)) &&
+				// forcing it be the specific arch and os (defalts to true, if not set)
+				osArchMatch(tags[i], opts.OS, opts.Architecture) {
```

Comment:
We are not testing os and arch for the regex case.

Let's start each loop with;

```go
if !osArchMatch(tags[i], opts) {
  continue
}
```